.github/workflows/benchmark-publish.yml

```diff
@@ -0,0 +1,23 @@
+name: benchmarks publish
```

Comment:
can this be merged to have a single github action; maybe by merging the `on:` triggers and adding a `if push + branch=master` condition?

Reply:
hmm, they are two different things, one is adding information about performance deltas on the PR, and the other is generating the dashboard with all the benchmark results. I can merge them if it clarifies what benchmark action does.